### PR TITLE
Hong Kong holidays update

### PR DESCRIPTION
**Pull request by KJhellico**
- holidays establishing years specified
- historical holidays added
- Lunar New Year and Christmas calculation fix
- Ching Ming Festival / Easter Monday overlapping fix
- optimizations
- tests added

### CODE REVIEW

File: holidays/countries/hongkong.py [hunk 1]
```diff
@@ -37,168 +35,140 @@
...
-                    self[new_year_date + rd(days=+1)] = second_day_lunar
-                    self[new_year_date + rd(days=+2)] = third_day_lunar
-                    self[new_year_date + rd(days=+3)] = fourth_day_lunar
         else:
             self[new_year_date] = name
             self[new_year_date + rd(days=+1)] = second_day_lunar
             self[new_year_date + rd(days=+2)] = third_day_lunar
 
+        easter_date = easter(year)
+        easter_monday_date = easter_date + rd(days=1)
```
[the remark this answers is not in the document]
Comment (KJhellico) [reply holidays/countries/hongkong.py:105]: I still don't know, how is more correct (or better), with `+` or without. So I completely rely on your opinion. ) If it's `days+N`, I will use it everywhere. Thank you!

File: holidays/countries/hongkong.py [hunk 2]
```diff
@@ -211,38 +181,29 @@ def _populate(self, year):
...
         # Chung Yeung Festival
         name = "Chung Yeung Festival"
-        dt = self.cnls.lunar_to_gre(year, 9, 9)
-        chung_yeung_date = date(dt.year, dt.month, dt.day)
-        if self.observed:
-            if chung_yeung_date.weekday() == SUN:
-                self[chung_yeung_date + rd(days=+1)] = day_following + name
-            else:
-                self[chung_yeung_date] = name
+        chung_yeung_date = self.cnls.lunar_to_gre(year, 9, 9)
```
[the remark this answers is not in the document]
Comment (KJhellico) [reply holidays/countries/hongkong.py:204]: But it's not September, it's 9th month of Chinese lunisolar calendar. So I think this constant is not applicable here.

File: holidays/countries/hongkong.py [hunk 3]
```diff
@@ -37,168 +35,140 @@
...
-            + "people's war of resistance against Japanese aggression"
-        )
-        if year == 2015:
-            self[date(year, SEP, 3)] = name
 
         # Chinese Mid-Autumn Festival
         name = "Chinese Mid-Autumn Festival"
-        dt = self.cnls.lunar_to_gre(year, 8, 15)
-        mid_autumn_date = date(dt.year, dt.month, dt.day)
+        mid_autumn_date = self.cnls.lunar_to_gre(year, 8, 15)
```
[the remark this answers is not in the document]
Comment (KJhellico) [reply holidays/countries/hongkong.py:171]: Same here. )

File: holidays/countries/hongkong.py [hunk 4]
```diff
@@ -37,168 +35,140 @@
...
         third_day_lunar = "The third day of Lunar New Year"
         fourth_day_lunar = "The fourth day of Lunar New Year"
-        dt = self.cnls.lunar_n_y_date(year)
-        new_year_date = date(dt.year, dt.month, dt.day)
+        new_year_date = self.cnls.lunar_n_y_date(year)
         if self.observed:
-            self[new_year_date] = name
-            if new_year_date.weekday() in [MON, TUE, WED, THU]:
+            if new_year_date.weekday() == SUN:
+                if year in (2006, 2007, 2010):
```
[the remark this answers is not in the document]
Comment (KJhellico) [reply holidays/countries/hongkong.py:85]: Ah, yes, set is optimal. I remember )

File: holidays/countries/hongkong.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (arkid15r) [reply holidays/countries/hongkong.py:204]: I see, makes sense.

**Review by arkid15r — COMMENTED** [review]
LGTM!

**Pull request merged**